doc/man3/RSA_sign.pod

```diff
@@ -46,8 +46,8 @@ B<rsa> is the signer's public key.
 
 =head1 RETURN VALUES
 
-RSA_sign() returns 1 on success.
-RSA_verify() returns 1 on successful verification.
+RSA_sign() returns 1 on success and 0 for failure.
```

Comment:
I think this is supposed to be true. It looks to me like the CAPI engine included in OpenSSL does not respect this and can return -1. Probably this is a bug in the CAPI engine.

Reply:
Thinking about it more we could perhaps "fix" this in `RSA_sign()` here:

````
return rsa->meth->rsa_sign(type, m, m_len, sigret, siglen, rsa);
````
i.e. regardless of whether the method returns <0 we could convert it to 0 anyway.